chronosphere/tfschema/trace_common.go

```diff
@@ -68,23 +68,20 @@ var TraceSpanFilterListSchema = &schema.Schema{
 			"duration":         TraceDurationFilterSchema,
 			"error":            TraceBoolFilterSchema,
 			// Note: this is the preferable form (singular) of this list field.
-			"tag": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Elem:     traceTagFilterSchema,
-			},
+			"tag": TraceTagFilterSchema,
 			// This field is still defined for error messaging to existing users, but is deprecated. Prefer tag.
-			"tags": {
-				Type:       schema.TypeList,
-				Optional:   true,
-				Elem:       traceTagFilterSchema,
-				Deprecated: "use tag instead",
-			},
+			"tags":       TraceTagFilterSchema,
```

Comment:
why this change -- this drops the `Deprecated` field

Reply:
yeah, I decided the type reuse was cleaner and that the later-stage error message would suffice

I noticed this because on the `main@HEAD` running `make generate` produced a non-empty diff... because I had shed the "shared type" (apparent in intschema's `main` generation function). This had a bit of a code smell in that it produced two separate types and would've required duplicating shared functions for the two nearly-identical input/return types (one corresponding to the variation having been marked deprecated)

Reply:
Ah I see, this helper will come in handy then:
https://github.com/chronosphereio/terraform-provider-chronosphere/blob/c1157696f91084738bf97456492457c8a5e96192/chronosphere/tfschema/notification_policy.go#L92

this will share the type, but add the deprecated message

Reply:
that looks like exactly what I needed! strangely, it. still seems to yield a new type?

```

+       Tags            []TraceSpanFilterTags `intschema:"tags,optional"`
+}
+
+type TraceSpanFilterTags struct {
+       Key          string              `intschema:"key,optional"`
+       NumericValue *TraceNumericFilter `intschema:"numeric_value,optional,list_encoded_object"`
+       Value        *TraceStringFilter  `intschema:"value,optional,list_encoded_object"`
 }
```

Anyway, this is not actually an issue: the updated business logic only consumes the preferred field and otherwise just checks for the emptiness of the deprecated list element (without inspecting individual elements) so ... 🤷